### PR TITLE
Allow literal numbers as instruction parameters.

### PR DESCRIPTION
**Pull request by chriseth**
Fixes https://github.com/powdr-org/powdr/issues/307

### CODE REVIEW

File: pilgen/src/lib.rs [hunk 1]
```diff
@@ -999,4 +1044,56 @@ pol constant p_reg_write_X_CNT = [1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0] + [0]*;
...
+    pub fn negative_for_unsigned() {
+        let source = r#"
+reg pc[@pc];
+reg fp;
+
+instr instro x: unsigned { pc' = pc + x }
+
+instro 9223372034707292161;
+"#;
+        compile::<GoldilocksField>(None, source).unwrap();
```
Comment (leonardoalt) [inline pilgen/src/lib.rs:1097]: What is this testing?

File: pilgen/src/lib.rs [hunk 2]
```diff
@@ -366,9 +371,34 @@ impl<T: FieldElement> ASMPILConverter<T> {
...
+                            }
+                        }
+                        Input::Literal(_, LiteralKind::SignedConstant) => {
+                            // TODO evaluate expression
+                            if let Expression::Number(n) = a {
+                                instruction_literal_arg.push(InstructionLiteralArg::Number(n));
+                            } else if let Expression::UnaryOperation(UnaryOperator::Minus, expr) = a
+                            {
+                                if let Expression::Number(n) = *expr {
+                                    instruction_literal_arg.push(InstructionLiteralArg::Number(-n));
```
Comment (leonardoalt) [inline pilgen/src/lib.rs:398]: What if this is -max, wouldn't it overflow?

Comment (chriseth) [reply pilgen/src/lib.rs:398]: It's a finite field element, it's not two's complement.

Comment (leonardoalt) [reply pilgen/src/lib.rs:398]: ok

File: pilgen/src/lib.rs (hunk 1 above)
Comment (chriseth) [reply pilgen/src/lib.rs:1097]: The number is so large that it falls into the "negative" part of the field. It fails at this assertion: https://github.com/powdr-org/powdr/pull/310/files#diff-e35bc36cfc5b345185d7dc9a1e99aa31abba8b28dc15bda76e3d658e61881281R385

It's really weird that one cannot test for panic messages in this framework.

File: pilgen/src/lib.rs [hunk 3]
```diff
@@ -743,7 +778,13 @@ impl<T: FieldElement> Register<T> {
...
 enum Input {
     Register(String),
-    Label(String),
+    Literal(String, LiteralKind),
+}
+
+enum LiteralKind {
+    Label,
+    SignedConstant,
+    UnsignedConstant,
```
[the remark this answers is not in the document]
Comment (chriseth) [reply pilgen/src/lib.rs:787]: The string is the name of the parameter. I think it doesn't really belong here. Yeah, all these structs need to be cleaned up at some point.

File: pilgen/src/lib.rs (hunk 1 above)
Comment (leonardoalt) [reply pilgen/src/lib.rs:1097]: ah ok

**Review by leonardoalt — APPROVED** [review]

**Pull request merged**